File: baml_language/sdk_tests/crates/swift/type_shapes/customizable/roundtrip_tests/TestClassRefs.swift
// Roundtrip coverage for `Baml.class_refs` — port of python_pydantic2
// `roundtrip_tests/test_class_refs.py`.
import XCTest
import Baml

final class TestClassRefs: XCTestCase {
    func test_class_refs_make_outer() throws {
        let o = try Baml.class_refs.make_outer(value: 5)
        XCTAssertEqual(o.inner.value, 5)
    }

    func test_class_refs_round_trip_inner() throws {
        let i = Baml.class_refs.Inner(value: 3)
        XCTAssertEqual(try Baml.class_refs.round_trip_inner(i: i), i)
    }

    func test_class_refs_round_trip_outer() throws {
        let o = Baml.class_refs.Outer(inner: Baml.class_refs.Inner(value: 9))
        XCTAssertEqual(try Baml.class_refs.round_trip_outer(o: o), o)
    }
}

File: baml_language/sdk_tests/crates/python_pydantic2/type_shapes/customizable/roundtrip_tests/test_class_refs.py
"""Roundtrip coverage for `baml_sdk.class_refs` — class composition."""

import baml_sdk  # noqa: F401  — initializes the BAML runtime
from baml_sdk.class_refs import (
    Inner,
    Outer,
    make_outer,
    round_trip_inner,
    round_trip_outer,
)


def test_class_refs_make_outer():
    o = make_outer(value=5)
    assert o.inner.value == 5


def test_class_refs_round_trip_inner():
    i = Inner(value=3)
    assert round_trip_inner(i=i) == i


def test_class_refs_round_trip_outer():
    o = Outer(inner=Inner(value=9))
    assert round_trip_outer(o=o) == o
